lib/data/wrapper.js

```diff
@@ -333,37 +333,41 @@ const data = {
         // backend to the same external backend
         // and for Azure if it is the same account since Azure copy outside
         // of an account is async
-        const locationTypeMatch = request.headers[locationHeader] ?
-        config.getLocationConstraintType(sourceLocationConstraintName) ===
-        config.getLocationConstraintType(request.headers[locationHeader])
-        : true;
-        if (locationTypeMatch &&
-        (sourceLocationConstraintType === 'aws_s3' ||
-        (sourceLocationConstraintType === 'azure' && config.isSameAzureAccount(
-        sourceLocationConstraintName, request.headers[locationHeader])))) {
-            const location = storeMetadataParams
-            .metaHeaders[locationHeader];
-            const objectGetInfo = dataLocator[0];
-            const externalSourceKey = objectGetInfo.key;
-            return client.copyObject(request, location,
-            externalSourceKey, sourceLocationConstraintName, log,
-            (error, objectRetrievalInfo) => {
-                if (error) {
-                    return cb(error);
-                }
-                const putResult = {
-                    key: objectRetrievalInfo.key,
-                    dataStoreName: objectRetrievalInfo.
-                        dataStoreName,
-                    dataStoreType: objectRetrievalInfo.
-                        dataStoreType,
-                    size: storeMetadataParams.size,
-                    dataStoreETag: objectGetInfo.dataStoreETag,
-                    start: objectGetInfo.start,
-                };
-                const putResultArr = [putResult];
-                return cb(null, putResultArr);
-            });
+        if (config.backends.data === 'multiple') {
+            const destLocationConstraintName =
+                destBackendInfo.getControllingLocationConstraint();
+            const locationTypeMatch =
+            config.getLocationConstraintType(sourceLocationConstraintName) ===
+            config.getLocationConstraintType(destLocationConstraintName);
+            if (locationTypeMatch &&
+            (sourceLocationConstraintType === 'aws_s3' ||
+            (sourceLocationConstraintType === 'azure' &&
+            config.isSameAzureAccount(sourceLocationConstraintName,
+            destLocationConstraintName)))) {
```

Comment:
please make this conditional a utility function that returns true or false.  also, please add unit tests for the different scenarios.